waiter/src/waiter/scheduler.clj

```diff
@@ -119,7 +119,10 @@
                                     (ss/try+
                                       {:success (f)}
                                       (catch #(contains? retry-status-codes (:status %)) e
-                                        (log/warn (str "Scheduler unavailable (Error code:" (:status e) ").") msg)
+                                        (log/warn (str "scheduler unavailable (error code:" (:status e) ").") msg)
+                                        (ss/throw+ e))
+                                      (catch ConnectException e
```

Comment:
Should we handle TimeoutException?  Why shouldn't we handle any exception at all in the same manner?

Reply:
Other exceptions might not be transient.

Reply:
Can we find out what exception occurs on timeouts and handle that as well?

Reply:
Okay

Reply:
Done.